src/editor/esphome-editor.ts

```diff
@@ -18,56 +24,131 @@ if (loc.protocol === "https:") {
 const wsUrl = wsLoc.href;
 
 @customElement("esphome-editor")
-export class ESPHomeEditor extends LitElement {
-  @property() public configuration!: string;
-
+class ESPHomeEditor extends LitElement {
   private editor?: monaco.editor.IStandaloneCodeEditor;
   private editorActiveWebSocket: WebSocket | null = null;
   private editorValidationScheduled = false;
   private editorValidationRunning = false;
   private editorActiveSecrets = false;
 
-  @query("main", true) private container!: HTMLElement;
+  @property() public fileName!: string;
+  @query("mwc-snackbar", true) private _snackbar!: Snackbar;
+  @query("editor-container", true) private container!: HTMLElement;
 
-  public getValue() {
-    return this.editor!.getModel()?.getValue();
+  createRenderRoot() {
+    return this;
   }
 
-  render() {
+  protected render() {
+    const isSecrets =
+      this.fileName === "secrets.yaml" || this.fileName === "secrets.yml";
+
     return html`
       <style>
-        ${editorStyles}
-        ${ESPHomeEditor.styles}
+        editor-container {
```

Comment:
Why is this inline and not using `static styles = css`

Reply:
it  stops working when the `createRenderRoot() { return this; } ` thing is added